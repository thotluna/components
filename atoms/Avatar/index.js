export { default } from './Avatar'
export { options } from './constants'
export { default as styles } from './Avatar.module.css'
