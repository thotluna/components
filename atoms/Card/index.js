export { default } from './Card'
export { options } from './constants'
export { default as styles } from './Card.module.css'
