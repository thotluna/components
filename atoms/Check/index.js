export { default } from './Check'
export { options } from './constants'
export { default as styles } from './Check.module.css'
