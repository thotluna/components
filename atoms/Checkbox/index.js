export { default } from './Checkbox'
export { options } from './constants'
export { default as styles } from './Checkbox.module.css'
