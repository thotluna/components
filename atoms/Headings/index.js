export { default, Headings } from './Headings'
export { options } from './constants'
export { default as styles } from './Headings.module.css'
