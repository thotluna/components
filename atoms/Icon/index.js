export { default } from './Icon'
export { options } from './constants'
export { default as styles } from './Icon.module.css'
