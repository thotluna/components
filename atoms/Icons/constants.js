import React from 'react'
export const options = {
  colors: ['base', 'primary', 'secondary', 'inverted', 'sun'],
  size: ['sm', 'md', 'lg'],
  type: [
    'chevron-left',
    'chevron-down',
    'check',
    'drag',
    'moon',
    'sun',
    'trash',
    'close',
  ],
}

export const getIcon = (icon) => {
  switch (icon) {
    case 'chevron-left':
      return (
        <svg
          xmlns="http://www.w3.org/2000/svg"
          fill="none"
          viewBox="0 0 24 24"
          stroke="currentColor"
          width="100%"
          height="100%"
        >
          <path
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth={1}
            d="M15 19l-7-7 7-7"
          />
        </svg>
      )
    case 'chevron-down':
      return (
        <svg
          xmlns="http://www.w3.org/2000/svg"
          fill="none"
          viewBox="0 0 24 24"
          stroke="currentColor"
          width="100%"
          height="100%"
        >
          <path
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth={1}
            d="M19 9l-7 7-7-7"
          />
        </svg>
      )
    case 'check':
      return (
        <svg
          xmlns="http://www.w3.org/2000/svg"
          fill="none"
          viewBox="0 0 24 24"
          stroke="currentColor"
          width="100%"
          height="100%"
        >
          <path
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth={1}
            d="M5 13l4 4L19 7"
          />
        </svg>
      )
    case 'trash':
      return (
        <svg
          xmlns="http://www.w3.org/2000/svg"
          fill="none"
          viewBox="0 0 24 24"
          stroke="currentColor"
          width="100%"
          height="100%"
        >
          <path
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth={1}
            d="M19 7l-.867 12.142A2 2 0 0116.138 21H7.862a2 2 0 01-1.995-1.858L5 7m5 4v6m4-6v6m1-10V4a1 1 0 00-1-1h-4a1 1 0 00-1 1v3M4 7h16"
          />
        </svg>
      )
    case 'drag':
      return (
        <svg
          xmlns="http://www.w3.org/2000/svg"
          fill="none"
          viewBox="0 0 24 24"
          stroke="currentColor"
          width="80%"
          height="80%"
        >
          <path
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth={1}
            d="M7 11.5V14m0-2.5v-6a1.5 1.5 0 113 0m-3 6a1.5 1.5 0 00-3 0v2a7.5 7.5 0 0015 0v-5a1.5 1.5 0 00-3 0m-6-3V11m0-5.5v-1a1.5 1.5 0 013 0v1m0 0V11m0-5.5a1.5 1.5 0 013 0v3m0 0V11"
          />
        </svg>
      )
    case 'moon':
      return (
        <svg
          xmlns="http://www.w3.org/2000/svg"
          fill="none"
          viewBox="0 0 24 24"
          stroke="currentColor"
          width="100%"
          height="100%"
        >
          <path
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="1"
            d="M20.354 15.354A9 9 0 018.646 3.646 9.003 9.003 0 0012 21a9.003 9.003 0 008.354-5.646z"
          />
        </svg>
      )
    case 'sun':
      return (
        <svg
          xmlns="http://www.w3.org/2000/svg"
          fill="none"
          viewBox="0 0 24 24"
          stroke="currentColor"
          width="100%"
          height="100%"
        >
          <path
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth={1}
            d="M12 3v1m0 16v1m9-9h-1M4 12H3m15.364 6.364l-.707-.707M6.343 6.343l-.707-.707m12.728 0l-.707.707M6.343 17.657l-.707.707M16 12a4 4 0 11-8 0 4 4 0 018 0z"
          />
        </svg>
      )
    case 'close':
      return (
        <svg
          xmlns="http://www.w3.org/2000/svg"
          viewBox="0 0 21 21"
          stroke="currentColor"
          width="100%"
          height="100%"
        >
          <g
            fill="none"
            fillRule="evenodd"
            strokeLinecap="round"
            strokeLinejoin="round"
          >
            <path d="M15.5 15.5l-10-10zM15.5 5.5l-10 10" />
          </g>
        </svg>
      )
  }
}
