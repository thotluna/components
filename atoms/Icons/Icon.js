import React from 'react'
import PropTypes from 'prop-types'
import styles from './Icon.module.css'
import { getClasses } from '../../helpers/styles'
import { getIcon, options } from './constants'

export const Icon = ({ type, color, size, hasBorder }) => {
  const classStyles = getClasses(styles)({ color, size })
  return (
    <div
      className={classStyles('container', ['color', 'size'], {
        'has-borde-base': hasBorder && color === 'base',
        'has-borde-primary': hasBorder && color === 'primary',
        'has-borde-secondary': hasBorder && color === 'secondary',
      })}
    >
      {getIcon(type)}
    </div>
  )
}

Icon.propTypes = {
  type: PropTypes.oneOf(options.type),
  color: PropTypes.oneOf(options.colors),
  hasBorder: PropTypes.bool,
  size: PropTypes.oneOf(options.size),
}

Icon.defaultProps = {
  color: 'sun',
  type: 'chevron-left',
  size: 'sm',
  hasBorder: false,
}

export default Icon
