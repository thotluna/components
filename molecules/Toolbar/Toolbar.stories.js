import { Toolbar, styles } from './index'
import { options as iconOptions } from '../../atoms/Icons'
import {
  // getListTemplate,
  getTemplate,
  getOptionsArgTypes,
} from '../../helpers/storybook'

const Template = getTemplate(Toolbar, styles)
// const ListTemplate = getListTemplate(Toolbar, styles)

export default {
  title: 'Molecules/Toolbar',
  component: Toolbar,
  args: {
    title: 'Login',
    icon: 'chevron-left',
  },
  argTypes: {
    title: { control: 'text' },
    icon: getOptionsArgTypes(iconOptions.type),
  },
}

export const Default = Template.bind({})

export const OnlyIcon = Template.bind({})
OnlyIcon.args = { icon: 'chevron-left', title: null }

export const OnlyTitle = Template.bind({})
OnlyTitle.args = { icon: null, title: 'Home' }
