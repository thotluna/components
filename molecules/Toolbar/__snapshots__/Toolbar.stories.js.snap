// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Molecules/Toolbar Default 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "background": "",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
      "padding": "2em",
    }
  }
>
  <div
    className="container"
  >
    <button
      aria-label="back"
      style={
        Object {
          "backgroundColor": "transparent",
          "border": 0,
          "padding": 0,
        }
      }
    >
      <div
        className="container color-inverted size-md"
      >
        <svg
          fill="none"
          height="100%"
          stroke="currentColor"
          viewBox="0 0 24 24"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15 19l-7-7 7-7"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth={1}
          />
        </svg>
      </div>
    </button>
    <div
      className="spacer vertical"
      style={
        Object {
          "height": "100%",
          "maxHeight": "auto",
          "width": 15,
        }
      }
    />
    <h2
      className="container color-inverted"
    >
      Login
    </h2>
  </div>
</div>
`;

exports[`Storyshots Molecules/Toolbar Only Icon 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "background": "",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
      "padding": "2em",
    }
  }
>
  <div
    className="container"
  >
    <button
      aria-label="back"
      style={
        Object {
          "backgroundColor": "transparent",
          "border": 0,
          "padding": 0,
        }
      }
    >
      <div
        className="container color-inverted size-md"
      >
        <svg
          fill="none"
          height="100%"
          stroke="currentColor"
          viewBox="0 0 24 24"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15 19l-7-7 7-7"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth={1}
          />
        </svg>
      </div>
    </button>
    <div
      className="spacer vertical"
      style={
        Object {
          "height": "100%",
          "maxHeight": "auto",
          "width": 15,
        }
      }
    />
  </div>
</div>
`;

exports[`Storyshots Molecules/Toolbar Only Title 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "background": "",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
      "padding": "2em",
    }
  }
>
  <div
    className="container"
  >
    <button
      aria-label="back"
      style={
        Object {
          "backgroundColor": "transparent",
          "border": 0,
          "padding": 0,
        }
      }
    />
    <div
      className="spacer vertical"
      style={
        Object {
          "height": "100%",
          "maxHeight": "auto",
          "width": 15,
        }
      }
    />
    <h2
      className="container color-inverted"
    >
      Home
    </h2>
  </div>
</div>
`;
